File: src/main/java/com/learning_platform/learning_resources/dto/RegisterCredentialsDTO.java
package com.learning_platform.learning_resources.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class RegisterCredentialsDTO {
    @Length(min=3)
    private String firstName;

    @Length(min=3)
    private String lastName;
    @Length(min=5)
    private String email;

    @Length(min=8)
    private String password;
    @Length(min=10)
    private String phoneNr;
    private String type;
}

File: app/model/RegisterCredentialsDTO.ts
export class RegisterCredentialsDTO{
  firstName:string;
  lastName:string;
  email:string;
  password: string;
  phoneNr:string;
  type:string;

  constructor(firstName:string,lastName:string,email:string,password:string,phoneNr:string,type:string) {
    this.lastName=lastName;
    this.firstName=firstName;
    this.email=email;
    this.password = password;
    this.phoneNr=phoneNr;
    this.type=type;
  }

}
